proposals/infra/INF-0001-availability-diagnostics.md

```diff
@@ -270,26 +294,26 @@ may produce warnings or errors with Clang.
 
 ### Diagnostic text
 
-The diagnostic message should reflect whether the intrisic is not available
-just for the particular target shader stage or the whole shader model version.
-In other words, it should be relevant to the entry point type. 
+The diagnostic message should reflect whether the function is not available for
+the shader model version or just for the specific shader stage. In other words,
+it should be relevant to the entry point type. 
 
-If intrinsic `a` is available in a shader model higher than the target shader
+If function `a` is available in a shader model higher than the target shader
 model regardless of target shader stage the diagnostic message should be:
 ```
-'a' is available beginning with Shader Model x.y
+'a' is only available on HLSL Shader Model x.y or newer
 ```
 
-If intrinsic `a` is not available in shader stage `S` regardless of shader model
+If function `a` is not available in shader stage `S` regardless of shader model
 version the diagnostic message should be:
 ```
-'a' is not available in S shaders
+'a' is not available in S shader environment on HLSL Shader Model x.y
 ```
 
-If intrinsic `a` is available in shader stage `S` in shader model higher than
-the target shader model the diagnostic message should be:
+If function `a` is available in shader stage `S` in shader model higher than the
+target shader model the diagnostic message should be:
 ```
-'a' is available in S shaders beginning with Shader Model x.y
+'a' is only available in S shader environment on HLSL Shader Model x.y or newer

```

Comment:
```suggestion
'a' is only available in 'S' shader environment on HLSL Shader Model x.y or newer

```

Wondering about quoting the name of the environment here to make it more clear what the variable thing is?

Reply:
While I'm picking at this, is "HLSL Shader Model" the correct term to use?  HLSL can be used in contexts where Shader Model doesn't apply, and also non-HLSL languages can be used with D3D where Shader Model does apply.

Reply:
> Wondering about quoting the name of the environment here to make it more clear what the variable thing is?

I'd prefer to leave it unquoted. It will resolve to "...in pixel shader environment..." and such and it is based on the compilation target. The 'a' comes from the input file so that's why it is quoted.